website/docs/timeline.md

```diff
@@ -3,40 +3,386 @@ title: Timeline
 
-* `Instant action` : Type of action performed on the table
-* `Instant time` : Instant time is typically a timestamp (e.g: 20190117010349), which monotonically increases in the order of action's begin time.
-* `state` : current state of the instant
+An instant can alter one or many partitions:
 
-Hudi guarantees that the actions performed on the timeline are atomic & timeline consistent based on the instant time.
+-   If you have one batch ingestion, you’ll see that as one commit in the active timeline. If you ingest streaming data, you might see multiple commits in the active timeline.When you open that commit file, you’ll see a JSON object with metadata about how one or more partitions were altered.
 
-Key actions performed include
+We’ll go over some details and concepts about the active and archived timeline below. All files in the timelines are immutable.
 
-* `COMMITS` - A commit denotes an **atomic write** of a batch of records into a table.
-* `CLEANS` - Background activity that gets rid of older versions of files in the table, that are no longer needed.
-* `DELTA_COMMIT` - A delta commit refers to an **atomic write** of a batch of records into a  MergeOnRead type table, where some/all of the data could be just written to delta logs.
-* `COMPACTION` - Background activity to reconcile differential data structures within Hudi e.g: moving updates from row based log files to columnar formats. Internally, compaction manifests as a special commit on the timeline
-* `ROLLBACK` - Indicates that a commit/delta commit was unsuccessful & rolled back, removing any partial files produced during such a write
-* `SAVEPOINT` - Marks certain file groups as "saved", such that cleaner will not delete them. It helps restore the table to a point on the timeline, in case of disaster/data recovery scenarios.
+:::caution
+The user should never directly alter the timeline (i.e. manually delete the commits).
+:::
 
-Any given instant can be
-in one of the following states
+## Active Timeline
 
-* `REQUESTED` - Denotes an action has been scheduled, but has not initiated
-* `INFLIGHT` - Denotes that the action is currently being performed
-* `COMPLETED` - Denotes completion of an action on the timeline
+The active timeline is a source of truth for all write operations: when an action (described below) happens on a table, the timeline is responsible for recording it. This guarantees a good table state, and Hudi can provide read/write isolation based on the timeline. For example, when data is being written to a Hudi table (i.e., requested, inflight), any data being written as part of the transaction is not visible to a query engine until the write transaction is completed. The query engine can still read older data, but the data inflight won’t be exposed.
```

Comment:
we should explain how timeline is a log, and atomicity is achieve by simply relying on atomic puts to storage.